app.go

```diff
@@ -79,14 +80,36 @@ func main() {
 }
 
 func write(handler, itemName, secret string) error {
-	// FIX: validate secret is valid by marshaling it
+	s := &responseStatus{}
```

Comment:
Hey, thanks for your PR!
Please could you move the validation into a seperate function and write some tests for it?
Thanks!